modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -84,12 +79,13 @@ func (k Keeper) OnChanOpenAck(
 		return sdkerrors.Wrapf(icatypes.ErrUnknownDataType, "cannot unmarshal ICS-27 interchain accounts metadata")
 	}
 
-	if err := icatypes.ValidateAccountAddress(metadata.Address); err != nil {
-		return err
+	channel, found := k.channelKeeper.GetChannel(ctx, portID, channelID)
+	if !found {
+		return sdkerrors.Wrapf(channeltypes.ErrChannelNotFound, "failed to retrieve channel %s on port %s", channelID, portID)
 	}
 
-	if metadata.Version != icatypes.Version {
-		return sdkerrors.Wrapf(icatypes.ErrInvalidVersion, "expected %s, got %s", icatypes.Version, metadata.Version)
+	if err := icatypes.ValidateControllerMetadata(ctx, k.channelKeeper, channel.ConnectionHops, metadata); err != nil {
```

Comment:
should the controller reject the metadata if the host didn't set the address?

Reply:
It's a good point. Although we can't really put it inside the `ValidateControllerMetadata` func as it won't be set on INIT, only on ACK.

Reply:
yea I'm thinking we should add a check in ACK that the address is non empty (outside validate controller metadata)

Reply:
Added an additional check and test case 👍